client/tests/simulator_tests.rs

```diff
@@ -83,6 +85,66 @@ fn test_get_capability_manufacturer_id() {
     assert_eq!(received, expected);
 }
 
+#[test]
+fn test_get_random_max_size() {
+    let (_sim_lifeline, mut tpm) = get_started_tpm();
+
+    let command = GetRandomCmd {
+        bytes_requested: 4096,
+    };
+
+    // We allow panic in test cases.
+    let resp = run_command(&command, &mut tpm).expect("Failed running command.");
+
+    // Lets pull out the actual data as a slice for convenience
+    let random_slice = &resp.random_bytes.buffer[..resp.random_bytes.size as usize];
+
+    // We should have received less than what the client implementation supports.
+    assert!(random_slice.len() <= TpmtHa::UNION_SIZE);
+
+    // We should have received minimum size of SHA256 (minimum mandatory digest size).
+    assert!(random_slice.len() >= TPM2_SHA256_DIGEST_SIZE as usize);
+
+    // Print for test inspection.
+    trace!("Got random sequence: {:?}", random_slice);
+
+    // Duplicate value trap, to catch any same value sequences.
+    {
+        let mut same_twice: u16 = 0;
+        for i in 1..random_slice.len() as usize {
+            if random_slice[i - 1] == random_slice[i] {
+                same_twice += 1;
+            }
+        }
+
+        trace!(
+            "After {} values, we got duplicate value {same_twice} times.",
+            random_slice.len()
+        );
+        assert!(same_twice <= random_slice.len() as u16 / 2); // Same value repeated in half of the values unlikely with > 32 values.
+    }
+}
+
+#[test]
+fn test_get_random_small_sizes() {
+    let (_sim_lifeline, mut tpm) = get_started_tpm();
+
+    for i in 0..4 as usize {
+        let command = GetRandomCmd {
+            bytes_requested: i as u16,
+        };
+
+        // We allow panic in test cases.
+        let resp = run_command(&command, &mut tpm).expect("Failed running command.");
+
+        // Lets pull out the actual data as a slice for convenience
+        let random_slice = &resp.random_bytes.buffer[..resp.random_bytes.size as usize];
+
+        // We should have received zero bytes.
+        assert!(random_slice.len() == i);
```

Comment:
Probably better to `assert_eq!` instead.

Reply:
Also, you probably don't need to form the slice, you could probably just inspect `resp.random_bytes.size` for this test.

Reply:
Fixed